### ISSUE TITLE
Spurious file ignored files warning running eslint in junctioned directory

### ISSUE THREAD

**Issue by marcstober** [eslint/eslint#4282]
**What version are you using?**
1.7.3, also able to reproduce in 1.5.1. On Windows.

I did *not* have this problem prior to upgrading from 1.2.0.

**What did you do?**
Ran eslint on a file in a junctioned directory.

For example:

```
C:\>d:

D:\>mkdir eslintbug

D:\>cd eslintbug

D:\eslintbug>touch foo.js

D:\eslintbug>eslint foo.js ### correct, no errors or warnings ###

D:\eslintbug>mklink /J c:\eslintbug d:\eslintbug
Junction created for c:\eslintbug <<===>> d:\eslintbug

D:\eslintbug>c:

C:\>cd eslintbug

C:\eslintbug>eslint foo.js

C:\eslintbug\foo.js
  0:0  warning  File ignored because of your .eslintignore file. Use --no-ignore to override 
### should not have this warning, it's the same file and no eslintignore even exists ###

? 1 problem (0 errors, 1 warning)
```

**What happened?**
"File ignored" warning on a file that wasn't (or at least shouldn't have been) ignored. See above.

**What did you expect to happen?**
Expected the file to be linted without this warning appearing.

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
Can you run eslint with `--debug` flag for the second case and provide an output? Is it possible that the directory you are linking to has an .eslintignore file (don't think that should matter, but just in case).

**Comment by marcstober** [comment]
Here's the debug output. There's no .eslintignore file.

```
C:\eslintbug>eslint --debug foo.js
  eslint:cli Running on files +0ms
  eslint:cli-engine Processing d:\eslintbug\foo.js +125ms
  eslint:cli-engine Linting d:\eslintbug\foo.js +3ms
  eslint:config Constructing config for d:\eslintbug\foo.js +1ms
  eslint:config Using .eslintrc and package.json files +2ms
  eslint:config Merging command line environment settings +4ms
  eslint:cli-engine Linting complete in: 21ms +5ms

C:\eslintbug\foo.js
  0:0  warning  File ignored because of your .eslintignore file. Use --no-ignore to override

? 1 problem (0 errors, 1 warning)
```

**Comment by nzakas** [comment]
This is most likely a duplicate of #3720.

**Comment by marcstober** [comment]
Yes, it does sound like #3720, I'd searched for a bug referencing a Windows "junction" but it's the same thing on a Unix link.

Looking at the code (https://github.com/eslint/eslint/blob/master/lib/cli-engine.js#L706) the warning is not actually coming from the code that's doing the ignoring; it looks like there's a separate iteration of files using a different mechanism to resolve paths, and then giving a warning if a file wasn't processed (as opposed to checking if it *was* ignored). I wonder if putting this warning closer to the logic that actually processes the .eslintignore would make a more reliable warning?

**Comment by nzakas** [comment]
Let's continue on #3720. A possible solution was discussed there.
